### PR TITLE
Services/nvdrv: add '/dev/nvhost-nvdec' device

### PR DESCRIPTION
**Pull request by mailwl**
This device used in game Monster Energy Supercross and some other

### CODE REVIEW

**Comment by Subv** [comment]
I would prefer it if we had a generic NvChannel class that implements the boilerplate ioctls and other channels just inherited from it (nvhost-gpu, nvhost-nvdec, nvhost-vic, etc)

**Comment by greggameplayer** [comment]
@mailwl if you want to put more Ioctls Channel here is the list of these : 
http://switchbrew.org/index.php?title=NV_services#Channel_Ioctls

File: src/core/hle/service/nvdrv/devices/nvhost_nvdec.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class nvhost_nvdec final : public nvdevice {
+public:
+    nvhost_nvdec() = default;
+    ~nvhost_nvdec() override = default;
+
+    u32 ioctl(Ioctl command, const std::vector<u8>& input, std::vector<u8>& output) override;
+
+private:
+    enum class IoctlCommand : u32_le {
```
Comment (lioncash) [inline src/core/hle/service/nvdrv/devices/nvhost_nvdec.h]: This will break on a big-endian system, as `u32_le` won't be a primitive arithmetic type.

**Comment by Hexagon12** [comment]
With this, 1-2-Switch now needs mm:u.

**Comment by bunnei** [comment]
Seems fine to me... will leave it up to someone to make a separate change refactoring with a generic NvChannel class.

**Pull request merged**